### PR TITLE
Only copy_v_cs_from_GPU if there are swimming particles

### LINKED ISSUE #2389: ENGINE: Un-needed data transfers
@hmenke @jdgraaf
In its current form, the engine feature copies v_cs data 
(velocities at particle center and 2nd point, I assume) 
from the gpu and distributes it via mpi, independently of whether swimming particles are there.
This takes some 20% of the execution time in an unrelated lbgpu+particle simulation.
19.97%  python         EspressoCore.so.4               [.] _ZN12_GLOBAL__N_18set_v_csEN5Utils5RangeI16ParticleIterator

we are currently preparing to extend espresso's default myconfig to include nearly all features, so we can ship pre-built packages for common Linux distros.

Unless above issue is resolved, we cannot include the ENGINE feature in this default config.
That would be a pity, because there is a tutorial for this feature.

### PR DESCRIPTION
**Pull request by hmenke**
Fixes #2389

Description of changes:
 - Add a global variable that indicates that there are swimming particles

### CODE REVIEW

**Comment by hmenke** [comment]
GitLab CI failed for `maxset` and `osx` in the `lb_thermostat` test (unsure whether this is related, it shouldn't be)
```
FAIL: test_velocity_distribution (__main__.LBCPUThermostat)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/testaccounticp/builds/espressomd/espresso/build/testsuite/python/lb_thermostat.py", line 77, in test_velocity_distribution
    self.assertLessEqual(abs(found - expected), error_tol)
AssertionError: 0.010366421182996383 not less than or equal to 0.01
```
The `rocm-maxset` job didn't start and failed with “No such container”.

**Review by mkuron — APPROVED** [review]

**Comment by mkuron** [comment]
Thermostat: unrelated, see #2386. rocm: job is being retried right now.

File: src/core/particle_data.cpp [hunk 1]
```diff
@@ -76,6 +76,7 @@ void add_id_to_type_map(int part_id, int type);
 
 int max_seen_particle = -1;
 int n_part = 0;
+bool has_swimming_particles = false;
```
Comment (KaiSzuttor) [inline src/core/particle_data.cpp]: What "has" swimming particles? Maybe I'm pedantic, but a name like ```swimming_particles_exist``` would make more sense in my opinion.

**Review by KaiSzuttor — APPROVED** [review]

**Comment by KaiSzuttor** [comment]
@RudolfWeeber

**Comment by RudolfWeeber** [comment]
Milestoning this for 4.1, since it only affects performance in custom myconfigs including  but not using ENGINE in 4.0.

**Comment by RudolfWeeber** [comment]
Please make activating the switch conditional on parameters that actually lead to swimming.
The current implementation will probably trigger even when a checkpoint is restored which does not contain swimming particles.

Once that's done, the pr can be merged.

**Pull request merged**